wire-ios-request-strategy/Sources/Synchronization/Decoding/StoreUpdateEvent.swift

```diff
@@ -95,6 +112,27 @@ public final class StoredUpdateEvent: NSManagedObject {
         return storedEvent
     }
 
+    static private func storedEventExists(for updateEventHash: Int, in context: NSManagedObjectContext) -> Bool {
+
+        let storedEvents = StoredUpdateEvent.events(eventHash: updateEventHash, context: context)
+        guard !storedEvents.isEmpty else {
+            return false
+        }
+
+        for storedEvent in storedEvents where storedEvent.eventHash == updateEventHash || storedEvent.eventHash == 0 {
```

Comment:
I don't quite understand the need for this check. Are there cases where storedEvents > 0 but we should return false? I was assuming we could just use [countForFetchRequest](https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506868-countforfetchrequest) here but I must be missing something.

Reply:
good idea, i replace it with a fetchRequest directly